eng/targets/Npm.Common.targets

```diff
@@ -24,7 +24,11 @@
 
   <Target Name="Restore">
     <Message Importance="High" Text="Running yarn install on $(MSBuildProjectFullPath)" />
-    <Yarn Command="install --mutex network $(InstallArgs)" StandardOutputImportance="High" StandardErrorImportance="High" />
+    <Yarn Command="install --mutex network $(InstallArgs)" StandardOutputImportance="High" StandardErrorImportance="High" IgnoreExitCode="true" >
+      <Output TaskParameter="ExitCode" PropertyName="YarnExitCode" />
+    </Yarn>
+
+    <Error Condition="'$(YarnExitCode)' != '0'" Text="Yarn exited with non-zero exit code: $(YarnExitCode). Ensure NodeJS is installed or exclude NodeJS with -NoBuildNodeJS (cmd), --no-build-nodejs (sh) or /p:BuildNodeJS=false (dotnet cli)." />
```

Comment:
<3 @Tratcher  eat your heart out

Reply:
Link to https://github.com/aspnet/AspNetCore/blob/master/docs/BuildFromSource.md?
How long does the build run before showing this error?

Reply:
This error shows up at restore time, is that okay?

Reply:
An error during restore doesn't sound :ok:

Reply:
TIL about the InitialTargets option on <Project>. Updated so now it fails before all other targets are run (restore, build, etc.).